Assets/Scripts/Sleutel.cs

```diff
@@ -0,0 +1,34 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class Sleutel : MonoBehaviour {
+
+	public GameObject deur;
+
+	private bool _triggered;
+	//private AudioSource _audioSource;
+	// Use this for initialization
+	void Awake () 
+	{
+		//_audioSource = gameObject.GetComponent<AudioSource>();
+	}
+	
+	void Update () 
+	{
+		if (_triggered ) //&& !_audioSource.isPlaying
+		{
+			Destroy (gameObject);
+			deur.SendMessage("GetKey");
+
+		}
+	}
+
+	void OnTriggerEnter2D(Collider2D collision)
+	{
+		if (collision.gameObject.tag == "Player") {
+			_triggered = true;
```

Comment:
Je kan de sendmessage en de destroy in de OnTrigger kwijt, dan hoef je de bool niet te checken in de update